src/server/snapshot.cc

```diff
@@ -390,7 +386,7 @@ void SliceSnapshot::OnJournalEntry(const journal::JournalItem& item, bool await)
   // To enable journal flushing to sync after non auto journal command is executed we call
   // TriggerJournalWriteToSink. This call uses the NOOP opcode with await=true. Since there is no
   // additional journal change to serialize, it simply invokes PushSerializedToChannel.
-  ConditionGuard guard(&bucket_ser_);
+  std::unique_lock lk(*db_slice_->GetThreadLocalMutex());
```

Comment:
Can we lock another lock here? Like a class-member (of `SliceSnapshot`) mutex?
I ask this because then:
1. This won't block the DbSlice, which is better in terms of reducing contention
2. We won't have to expose the internal mutex via `GetThreadLocalMutex()`, which is better encapsulation

Reply:
I think not, because it won't synchronize with the lock in `Traverse()` which is the point of this PR